content/backend/typescript-apollo/9-deployment.md

```diff
@@ -0,0 +1,692 @@
+```
+</Instruction>
+
+This is the format used by Prisma for connection strings:
+
+![Prisma Connection String format](https://i.imgur.com/qxnUKrS.png)
+
+Note that the **Arguments** portion is optional and is absent in your connection string. 
+
+This format is based on the [official PostgreSQLl format for connection URLs](https://www.postgresql.org/docs/current/libpq-connect.html#libpq-connstring). You can find more information about this in the [Prisma docs](https://www.prisma.io/docs/concepts/database-connectors/postgresql). 
+
+Now it's time to create migrations again, but this time for your new PostgreSQL database. 
+
+<Instruction>
+
+Use the `prisma` CLI tool to create a new migration:
+
+```bash(path=".../hackernews-typescript/")
+npx prisma migrate dev --name init
+```
+
+</Instruction>
+
+We have already covered the core concept behind the `migrate` command in [Chapter 4](../4-adding-a-database/). This command will also regenerate Prisma Client based on the most recent schema. 
+
+
+You can now restart server with `npm run dev`. While your previous data will have disappeared, all functionality should remain intact. Feel free to test out all the queries/mutations to see if they work as expected. 
+
+<Instruction>
+
+Add all your changes and create a new commit: 
+
+```bash(path=".../hackernews-typescript/")
+git add .
+git commit -m "migrate database to postgres"
+```
+
+</Instruction>
+
+### Update `index.ts` to make it suitable for deployment
+
+You will need to make a few changes to the `index.ts` to make the API ready for deployment. 
+
+<Instruction>
+
+
+Update `src/index.ts` as follows:
+
+```typescript{2,10-11,14}(path=".../hackernews-typescript/src/index.ts")
+import { ApolloServer } from "apollo-server";
+import { ApolloServerPluginLandingPageLocalDefault } from "apollo-server-core";
+
+import { schema } from "./schema";
+import { context } from "./context";
+
+export const server = new ApolloServer({
+    schema,
+    context,
+    introspection: true,                                      // 1
+    plugins: [ApolloServerPluginLandingPageLocalDefault()],   // 2
+});
+
+const port = process.env.PORT || 3000;                        // 3
+
+server.listen({ port }).then(({ url }) => {
+    console.log(`🚀  Server  ready at ${url}`);
+});
+
+```
+
+
+</Instruction>
+
+Let's explain the changes: 
+
+1. "[Introspection](https://graphql.org/learn/introspection/)" is a feature that allows a GraphQL client to ask a GraphQL server for information about what queries it supports. Apollo turns off introspection in production by default as a security measure ([more info](https://www.apollographql.com/blog/graphql/security/why-you-should-disable-graphql-introspection-in-production/#:~:text=Turning%20off%20introspection%20in%20production,%2C%20resolvers%2C%20introspection%3A%20process.)). However, we decided to keep introspection on for convenience, though this is not something you would normally do in a production application.
```

Comment:
```suggestion
1. "[Introspection](https://graphql.org/learn/introspection/)" is a feature that allows a GraphQL client to ask a GraphQL server for information about what queries it supports. Apollo turns off introspection in production by default as a security measure. However, we decided to keep introspection on for convenience, though this is not something you normally do in a production application.

>  Read further on [Why You Should Disable GraphQL Introspection In Production – GraphQL Security](https://www.apollographql.com/blog/graphql/security/why-you-should-disable-graphql-introspection-in-production/).
```
You can rephrase the wording here.

>However, we decided to keep introspection on for convenience, though this is not something you would normally do in a production application.

What convenience?

Reply:
Good point! Clarified.